src/schematools/contrib/django/management/commands/create_views.py

```diff
@@ -117,50 +135,93 @@ def create_views(
         for table in dataset.schema.tables:
             if table.is_view:
                 command.stdout.write(f"* Creating view {table.db_name}")
-                view_sql = _clean_sql(dataset.schema.get_view_sql())
-                if not _is_valid_sql(view_sql):
-                    command.stderr.write(f"  Invalid SQL for view {table.db_name}")
-                    errors += 1
-                    continue
                 required_permissions = _get_required_permissions(table)
                 view_dataset_auth = dataset.schema.auth
                 if _check_required_permissions_exist(view_dataset_auth, required_permissions):
                     try:
                         with connection.cursor() as cursor:
 
+                            # Check if write role exists and create if it does not
+                            write_role_name = f"write_{table._parent_schema.db_name}"
+                            _create_role_if_not_exists(cursor, write_role_name)
+
+                            # Grant usage and create on schema public to write role
+                            cursor.execute(
+                                sql.SQL(
+                                    "GRANT usage,create on schema public TO {write_role_name}"
+                                ).format(write_role_name=sql.Identifier(write_role_name))
+                            )
+
+                            # Check if the provides view_sql is correct
+                            view_sql = _clean_sql(dataset.schema.get_view_sql())
+                            if not _is_valid_sql(view_sql, table.db_name, write_role_name):
+                                command.stderr.write(f"  Invalid SQL for view {table.db_name}")
+                                continue
```

Comment:
Misschien goed om een comment bij deze regel te zetten? Is makkelijk over het hoofd te zien